platforms/posix/cmake/sitl_target.cmake

```diff
@@ -160,6 +161,7 @@ set(models
 	plane_catapult
 	plane_lidar
 	px4vision
+	quadx
 	r1_rover
```

Comment:
Since these models only work with SIH and none of the other simulators, could we not instantiate the make targets for other simulations?

Reply:
@romain-chiap You need to iterate through your sih's own list of models. You can look at jsbsim as an example in the same file

Reply:
This is true for jmavsim too, when typing `make px4_sitl list_vmd_make_targets` is shows me results like `jmavsim_plane_catapult_callgrind` which I'm quite sure do not exist.

Reply:
That is true, and we need to do this for jmavsim too. It would be better to do it properly whenever we have the opportunity

Reply:
I wasn't able to properly checkout this current PR, so I pushed it in "sih in sitl" branch PR-[18964](https://github.com/PX4/PX4-Autopilot/pull/18964/commits/9b43fb1d09acfee0d7826915b7b27638940a1225). Let me know if that's what you want.

Reply:
@Jaeyoung-Lim can you mark that as resolved?

Reply:
I did it in this other [branch](https://github.com/PX4/PX4-Autopilot/pull/18964/commits/9b43fb1d09acfee0d7826915b7b27638940a1225)
@ThomasDebrunner can you cherry-pick it?